Informatics II/001/Task 2/Yuirike Solution/Task2-GeneralCase.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+/*
+Test this code for general n x n cases
+If you use VSCode, you can run this with Code-Runner.
+*/
+
+//Function to calculate the dot product
+int dotP(int A[], int B[], int n)
+{
+    int res = 0;
+    for (int i = 0; i < n; i++)
+    {
+        for (int j = 0; j < n; j++)
+        {
+            if (i == j)
+            {
+                res = res + B[j] * A[i];
+            }
+        }
+    }
+    return res;
+};
+
+
+//Example case with a 4x4 matrix
+int main(void)
+{
+    int A[4][4] = {{1, 2, 4, 6}, {3, 4, 2, 6}, {1, 2, 4, 6}, {1, 2, 4, 6}};
+    int B[4][4];
+    int n = 4;
```

Comment:
You can put n as the size of the array immediately, so its easier to change the dimensions of the Matrix:
```suggestion

    int n = 4;
    int A[n][n] = {{1, 2, 4, 6}, {3, 4, 2, 6}, {1, 2, 4, 6}, {1, 2, 4, 6}};
    int B[n][n];    
```

Reply:
I also think that is a good idea but any idea what this one is about? 
![image](https://user-images.githubusercontent.com/116112098/223157866-ac313b21-fe66-4f0b-ba8d-898754058f4b.png)

Reply:
Probably because you hard code the "A" 2d-array with 4x4 values, the compiler notices a possible bug when n isn't 4.

Reply:
So you want me to leave it empty and just give the user the ability to make an array?

Reply:
Maybe also scan and condense both into one general script?

Reply:
(scan inside a loop, n times)

Reply:
> Maybe also scan and condense both into one general script?

I don't really understand what you want me to do.
Do you want me to modify the main script, the one solely completing Task 2 provided by the Ex01, such that the user can also choose the matrix size and then customize the array themselves? 

Or do you want me to change stuff within the script that is just provided as an extra to showcase how the functionality also works for n x n matrices?

Reply:
combine both into one, where the user can choose the size, give inputs and then it prints everything

Reply:
> combine both into one, where the user can choose the size, give inputs and then it prints everything

Changed it as requested, deleted the previous two files and "fused" them.
Note: I wasn't sure what kind of prompts I should print for the user, so I hope my clumsy choice of words isn't too bad.
Note: I wondered if it is possible to make the input easier, as now the user has to enter each digit individually. In Python, you could just let the user enter a string in a provided format and then extract the numbers like that. I assume that might be possible in C but trickier and probably too much to ask for. I hope this submission is ok.

Reply:
Perfect 👍